common/scala/src/main/scala/whisk/connector/lean/LeanProducer.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.connector.lean
+
+import scala.concurrent.Future
+import org.apache.kafka.clients.producer.RecordMetadata
+import whisk.common.Counter
+import whisk.common.Logging
+import whisk.core.connector.Message
+import whisk.core.connector.MessageProducer
+
+import java.util.concurrent.{BlockingQueue, ConcurrentMap, LinkedBlockingQueue}
+
+class LeanProducer(queues: ConcurrentMap[String, BlockingQueue[Array[Byte]]])(implicit logging: Logging)
+    extends MessageProducer {
+
+  override def sentCount(): Long = sentCounter.cur
+
+  /** Sends msg to topic. This is an asynchronous operation. */
+  override def send(topic: String, msg: Message, retry: Int = 3): Future[RecordMetadata] = {
+    implicit val transid = msg.transid
+
+    logging.debug(this, s"sending to topic '$topic' msg '$msg'")
+    var queue = queues.get(topic)
+    try {
+      queue.put(msg.serialize.getBytes("UTF-8"))
+      sentCounter.next()
+    } catch {
+      //TODO: shouldn't get here
+      case npe: NullPointerException => {
+        queue = new LinkedBlockingQueue[Array[Byte]]()
+        queues.put(topic, queue)
+        queue.put(msg.serialize.getBytes("UTF-8"))
+      }
+    }
+    Future.successful(null)
```

Comment:
MessageProducer trait defines send method to implement and it must return Future[RecordMetadata].
```
def send(topic: String, msg: Message, retry: Int = 0): Future[RecordMetadata]
````
The only thing i can think of without changing interface is to define sort of dummy RecordMetadata object in utils for cases like this and use it instead of null.
Maybe, we should consider to remove Kafka dependency from the interface and replace by some internal class?